react/components/Link.tsx

```diff
@@ -25,6 +25,17 @@ interface Props extends NavigateOptions {
   waitToPrefetch?: number
 }
 
+const appendWorkspace = (
+  url: string | undefined,
+  workspace: string | undefined
+) => {
+  if (!url || !workspace || workspace === '') {
+    return url
+  }
+
+  const separator = String(url).includes('?') ? '&' : '?'
+  return url + separator + 'workspace=' + workspace
```

Comment:
What if there is already a workspace query string? 👀

Reply:
what if indeed 👀